### PR TITLE
Static Analysis

### PR DESCRIPTION
**Pull request by stephenmichaelf**
Add Static Analysis to the Agent.

### CODE REVIEW

File: src/rules.ruleset [hunk 1]
```diff
@@ -0,0 +1,210 @@
+<?xml version="1.0"?>
+<RuleSet Name="All Rules Enabled with default severity" Description="All Rules are enabled with default severity. Rules with IsEnabledByDefault = false are force enabled with default severity." ToolsVersion="15.0">
+   <Rules AnalyzerId="Microsoft.CodeAnalysis.FxCopAnalyzers" RuleNamespace="Microsoft.CodeAnalysis.FxCopAnalyzers">
```
Comment (damccorm) [inline src/rules.ruleset]: How was this ruleset chosen?

File: .vsts.template.nonwindows.yml [hunk 2]
```diff
@@ -18,6 +19,8 @@ steps:
 - script: ./dev.sh layout Release
   workingDirectory: src
   displayName: Build & Layout Release
+  env:
```
Comment (damccorm) [inline .vsts.template.nonwindows.yml]: Could you give an example of this working?

Comment (jtpetty) [reply .vsts.template.nonwindows.yml]: https://dev.azure.com/mseng/PipelineTools/_build/results?buildId=11504349&view=logs&j=5b4209a6-3453-5624-9b5c-f3d75dcd52a3&t=06a06f86-759a-5666-a10d-c4678e42ef5f

File: src/rules.ruleset (hunk 1 above)
Comment (damccorm) [reply src/rules.ruleset]: Do we want to turn the ones that currently aren't being hit to Errors?

Comment (stephenmichaelf) [reply src/rules.ruleset]: We took the global ruleset that Microsoft suggests.

I think they are almost all being hit :) We don't have a list of cross referencing failures vs. what exists.

Comment (damccorm) [reply src/rules.ruleset]: That's fair, generating that sounds like more trouble than it would be worth

**Comment by jtpetty** [comment]
The story was just to get SA up and running and reporting results. Fixing the warnings will be a later effort.

**Comment by stephenmichaelf** [comment]
I created an issue [here](https://github.com/microsoft/azure-pipelines-agent/issues/2794) to track it.

**Pull request merged**